### PR TITLE
Handle channel exceptions in example client

### PR DESCRIPTION
**Pull request by Raqbit**
Solves #390.

The client example is now able to handle channel exceptions and will recreate the channel once one occurs.

### CODE REVIEW

File: example_client_test.go [hunk 1]
```diff
@@ -36,28 +36,32 @@ func Example() {
...
+
 	// When resending messages the server didn't confirm
 	resendDelay = 5 * time.Second
 )
 
 var (
 	errNotConnected  = errors.New("not connected to the queue")
-	errNotConfirmed  = errors.New("message not confirmed")
 	errAlreadyClosed = errors.New("already closed: not connected to the queue")
+	errShutdown      = errors.New("queue is shutting down")
```
Comment (michaelklishin) [inline example_client_test.go]: This is a good chance to correct this poorly worded message. Clients do not connect to queues, they [connect to nodes](https://www.rabbitmq.com/connections.html).

File: example_client_test.go [hunk 2]
```diff
@@ -36,28 +36,32 @@ func Example() {
 
 // Queue represents a connection to a specific queue.
```
Comment (michaelklishin) [inline example_client_test.go]: I know it's not in the original scope but this badly needs renaming. Clients do not connect to queues. `ConsumerState` would be a better name here.

File: example_client_test.go [hunk 3]
```diff
@@ -36,28 +36,32 @@ func Example() {
...
+	notifyConfirm   chan amqp.Confirmation
+	isReady         bool
 }
 
 const (
 	// When reconnecting to the server after connection failure
 	reconnectDelay = 5 * time.Second
 
+	// When setting up the channel after a channel exception
+	resetupDelay = 2 * time.Second
```
Comment (michaelklishin) [inline example_client_test.go]: "resetup" is not really a word. I think "reinitialization" or "reinit" would be clearer and more familiar here.

File: example_client_test.go [hunk 4]
```diff
@@ -73,35 +77,86 @@ func New(name string, addr string) *Queue {
...
-		return false
+		return nil, err
+	}
+
+	queue.changeConnection(conn)
+	log.Println("Connected!")
+	return conn, nil
+}
+
+func (queue *Queue) handelResetup(conn *amqp.Connection) bool {
```
Comment (michaelklishin) [inline example_client_test.go]: Same as above, and `handel` should probably be `handle`.

File: example_client_test.go [hunk 5]
```diff
@@ -73,35 +77,86 @@ func New(name string, addr string) *Queue {
...
+	for {
+		queue.isReady = false
+
+		err := queue.setup(conn)
```
Comment (michaelklishin) [inline example_client_test.go]: `ConsumerState.init` or `AppState.init` would fit quite well here.

File: example_client_test.go (hunk 2 above)
Comment (Raqbit) [reply example_client_test.go]: What do you think of `Session`?

Comment (michaelklishin) [reply example_client_test.go]: `Session` is pretty good.

**Comment by michaelklishin** [comment]
@Raqbit thank you for updating the names and being so patient.

**Pull request merged**